Realm.Shared/native/NativeException.cs

```diff
@@ -29,22 +29,22 @@ internal unsafe struct NativeException
         public sbyte* messageBytes;
         public IntPtr messageLength;
 
-        internal Exception Convert()
+        internal Exception Convert(Func<RealmExceptionCodes, Exception> overrider)
```

Comment:
This could also default to null? Or do you prefer explicitly requiring people pass in a null `overrider`?